libraries/WiFi/src/WiFiServer.h

```diff
@@ -24,6 +24,10 @@
 #include "WiFiClient.h"
 #include "IPAddress.h"
 
+#ifndef SERVER_MAX_MONITORED_CLIENTS
+#define SERVER_MAX_MONITORED_CLIENTS 5
```

Comment:
`CONFIG_LWIP_MAX_SOCKETS` is `16`, meaning there can be more than 5 monitored clients. What happens with the rest?